google_permissions/versions.tf

```
@@ -0,0 +1,9 @@
+terraform {
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = ">= 3.0"
+    }
+  }
+  required_version = "~> 1.0"
```

```suggestion
  required_version = ">= 1.2"
```
Since precondition is only available in 1.2.